test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -10724,7 +10727,12 @@ TEST_F(AtenXlaTensorTest, TestNanToNum) {
     ForEachDevice([&](const torch::Device& device) {
       torch::Tensor xla_input = CopyToDevice(input, device);
       torch::Tensor xla_output = torch::nan_to_num(xla_input);
-      AllClose(output, xla_output);
```

Comment:
Instead of doing this, you should use `std::numeric_limits<float>::infinity()` when it is on TPU, you can check that by 

```
bridge::AtenDeviceToXlaDevice(device).hw_type == DeviceType::TPU
```

Reply:
`std::numeric_limits<float>::infinity()` leads to a `double` vs. `float` runtime type error. 

As discussed offline, since TPU converts `double` to `float` (unlike CPU), we cannot fully test  condition. I limited this condition to TPU tests and added a clarification comment to the test.

Reply:
This part lgtm. After testing the float64 on TPU we can come back and remove this hack.